<a href="https://colab.research.google.com/github/DaveChui/Prep_Group-4-Project/blob/main/DS14_GROUP_IV_PROJECT_2022_02_CRYPTO_ANALYSIS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1.0 Importing the Libraries.

In [245]:
# Importing the pandas library
import pandas as pd

# Importing the numpy library
import numpy as np

#2.0 Data Loading and Cleaning.

In [246]:
#First load the bitcoin data and preview records 
bitcoin_data = pd.read_csv('/content/bitcoin.csv',delimiter=';')
bitcoin_data.head()

Date,Open,High,Low,Close,Adj Close,Volume
0  2017-11-21,8205.740234,8348.660156,7762.709961...
1  2017-11-22,8077.950195,8302.259766,8075.470215...
2  2017-11-23,8232.379883,8267.400391,8038.770020...
3  2017-11-24,8074.020020,8374.160156,7940.930176...
4  2017-11-25,8241.709961,8790.919922,8191.149902...

In [247]:
#Load the ethereum data and preview records 
ethereum_data = pd.read_csv('/content/ethereum.csv',delimiter=';')
ethereum_data.head()

Date,Open,High,Low,Close,Adj Close,Volume
0  2017-11-21,367.442993,372.470001,350.692993,36...
1  2017-11-22,360.312012,381.420013,360.147003,38...
2  2017-11-23,381.438995,425.548004,376.088013,41...
3  2017-11-24,412.501007,480.972992,402.757996,47...
4  2017-11-25,475.675995,485.191986,461.053009,46...

In [248]:
#Get information about our datasets
bitcoin_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 1 columns):
 #   Column                                     Non-Null Count  Dtype 
---  ------                                     --------------  ----- 
 0   Date,Open,High,Low,Close,Adj Close,Volume  1554 non-null   object
dtypes: object(1)
memory usage: 12.3+ KB


In [249]:
ethereum_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 1 columns):
 #   Column                                     Non-Null Count  Dtype 
---  ------                                     --------------  ----- 
 0   Date,Open,High,Low,Close,Adj Close,Volume  1554 non-null   object
dtypes: object(1)
memory usage: 12.3+ KB


In [250]:
#separating the columns in the bitcoin data
bitcoin_data = pd.read_csv('/content/bitcoin.csv', sep = ',')
bitcoin_data.head()

Date         Open         High  ...        Close    Adj Close      Volume
0  2017-11-21  8205.740234  8348.660156  ...  8071.259766  8071.259766  4277609984
1  2017-11-22  8077.950195  8302.259766  ...  8253.549805  8253.549805  3633530112
2  2017-11-23  8232.379883  8267.400391  ...  8038.770020  8038.770020  4225179904
3  2017-11-24  8074.020020  8374.160156  ...  8253.690430  8253.690430  5058610176
4  2017-11-25  8241.709961  8790.919922  ...  8790.919922  8790.919922  4342060032

[5 rows x 7 columns]

In [251]:
ethereum_data = pd.read_csv('/content/ethereum.csv', sep = ',')
ethereum_data.head()

Date        Open        High  ...       Close   Adj Close      Volume
0  2017-11-21  367.442993  372.470001  ...  360.401001  360.401001   949912000
1  2017-11-22  360.312012  381.420013  ...  380.652008  380.652008   800819008
2  2017-11-23  381.438995  425.548004  ...  410.165985  410.165985  1845680000
3  2017-11-24  412.501007  480.972992  ...  474.911011  474.911011  2292829952
4  2017-11-25  475.675995  485.191986  ...  466.276001  466.276001  1422080000

[5 rows x 7 columns]

In [252]:
#Get information about the dataset again to see if changes are effective.
bitcoin_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       1554 non-null   object 
 1   Open       1554 non-null   float64
 2   High       1554 non-null   float64
 3   Low        1554 non-null   float64
 4   Close      1554 non-null   float64
 5   Adj Close  1554 non-null   float64
 6   Volume     1554 non-null   int64  
dtypes: float64(5), int64(1), object(1)
memory usage: 85.1+ KB


In [253]:
#Get information about the dataset again to see if changes are effective.
ethereum_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       1554 non-null   object 
 1   Open       1554 non-null   float64
 2   High       1554 non-null   float64
 3   Low        1554 non-null   float64
 4   Close      1554 non-null   float64
 5   Adj Close  1554 non-null   float64
 6   Volume     1554 non-null   int64  
dtypes: float64(5), int64(1), object(1)
memory usage: 85.1+ KB


From the above info we'll change the Date column from object data type to date/time.

In [254]:
#Changing the Date column's data type from object to date/time.
bitcoin_data['Date'] = pd.to_datetime(bitcoin_data['Date'])
bitcoin_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   Date       1554 non-null   datetime64[ns]
 1   Open       1554 non-null   float64       
 2   High       1554 non-null   float64       
 3   Low        1554 non-null   float64       
 4   Close      1554 non-null   float64       
 5   Adj Close  1554 non-null   float64       
 6   Volume     1554 non-null   int64         
dtypes: datetime64[ns](1), float64(5), int64(1)
memory usage: 85.1 KB


In [255]:
#Changing the Date column's data type from object to date/time.
ethereum_data['Date'] = pd.to_datetime(ethereum_data['Date'])
ethereum_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   Date       1554 non-null   datetime64[ns]
 1   Open       1554 non-null   float64       
 2   High       1554 non-null   float64       
 3   Low        1554 non-null   float64       
 4   Close      1554 non-null   float64       
 5   Adj Close  1554 non-null   float64       
 6   Volume     1554 non-null   int64         
dtypes: datetime64[ns](1), float64(5), int64(1)
memory usage: 85.1 KB


In [256]:
#Ensure uniformity by changing the column names to lower case, stripping of the whitespaces,removing periods, 
#this ensures good readerbility plus every naming issue is covered.
#This is to be applied to both the bitcoin and ethereum datasets.
#Bitcoin dataset

bitcoin_data.columns = bitcoin_data.columns.str.lower().str.strip().str.replace(' ', '_').str.replace('.', '')
bitcoin_data.columns

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  


Index(['date', 'open', 'high', 'low', 'close', 'adj_close', 'volume'], dtype='object')

In [257]:
#Ethereum dataset
ethereum_data.columns = ethereum_data.columns.str.lower().str.strip().str.replace(' ', '_').str.replace('.', '')
ethereum_data.columns

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  


Index(['date', 'open', 'high', 'low', 'close', 'adj_close', 'volume'], dtype='object')

In [258]:
#Renaming the columns in both our datasets to differentiate between the bitcoin and ethereum columns by adding an 'eth' suffix for ethereum
#and a 'bit' for bitcoin.
bitcoin_data2 = bitcoin_data.add_suffix('bit')


In [259]:
#repeat the process for ethereum dataset

ethereum_data2 =ethereum_data.add_suffix('eth')

In [260]:
bitcoin_data2.head()

datebit      openbit      highbit  ...     closebit  adj_closebit   volumebit
0 2017-11-21  8205.740234  8348.660156  ...  8071.259766   8071.259766  4277609984
1 2017-11-22  8077.950195  8302.259766  ...  8253.549805   8253.549805  3633530112
2 2017-11-23  8232.379883  8267.400391  ...  8038.770020   8038.770020  4225179904
3 2017-11-24  8074.020020  8374.160156  ...  8253.690430   8253.690430  5058610176
4 2017-11-25  8241.709961  8790.919922  ...  8790.919922   8790.919922  4342060032

[5 rows x 7 columns]

In [261]:
ethereum_data2.head()

dateeth     openeth     higheth  ...    closeeth  adj_closeeth   volumeeth
0 2017-11-21  367.442993  372.470001  ...  360.401001    360.401001   949912000
1 2017-11-22  360.312012  381.420013  ...  380.652008    380.652008   800819008
2 2017-11-23  381.438995  425.548004  ...  410.165985    410.165985  1845680000
3 2017-11-24  412.501007  480.972992  ...  474.911011    474.911011  2292829952
4 2017-11-25  475.675995  485.191986  ...  466.276001    466.276001  1422080000

[5 rows x 7 columns]

In [262]:
#renaming the 'dateeth' column to 'datebit' to be able to merge the two datasets since the data was collected in the same period.

ethereum_data3 = ethereum_data2.rename(columns={'dateeth': 'datebit'}, inplace=True)


In [263]:
ethereum_data2

datebit      openeth  ...  adj_closeeth    volumeeth
0    2017-11-21   367.442993  ...    360.401001    949912000
1    2017-11-22   360.312012  ...    380.652008    800819008
2    2017-11-23   381.438995  ...    410.165985   1845680000
3    2017-11-24   412.501007  ...    474.911011   2292829952
4    2017-11-25   475.675995  ...    466.276001   1422080000
...         ...          ...  ...           ...          ...
1549 2022-02-17  3126.858398  ...   2881.481934  15860206214
1550 2022-02-18  2884.340576  ...   2785.727539  15748173433
1551 2022-02-19  2784.872803  ...   2763.701172   9774183169
1552 2022-02-20  2763.756592  ...   2628.648438  11641437834
1553 2022-02-21  2620.029541  ...   2701.658936  15016932352

[1554 rows x 7 columns]

#Merging the datasets

In [265]:
# creating a new data set by combining the Bitcoin and Ethereum datasets and displaying it.

crypto_data = pd.merge(bitcoin_data2, ethereum_data2, left_on='datebit', right_on='datebit')
crypto_data

datebit       openbit  ...  adj_closeeth    volumeeth
0    2017-11-21   8205.740234  ...    360.401001    949912000
1    2017-11-22   8077.950195  ...    380.652008    800819008
2    2017-11-23   8232.379883  ...    410.165985   1845680000
3    2017-11-24   8074.020020  ...    474.911011   2292829952
4    2017-11-25   8241.709961  ...    466.276001   1422080000
...         ...           ...  ...           ...          ...
1549 2022-02-17  43937.070313  ...   2881.481934  15860206214
1550 2022-02-18  40552.132813  ...   2785.727539  15748173433
1551 2022-02-19  40026.023438  ...   2763.701172   9774183169
1552 2022-02-20  40118.101563  ...   2628.648438  11641437834
1553 2022-02-21  38339.218750  ...   2701.658936  15016932352

[1554 rows x 13 columns]

In [266]:
#Checking the info again
crypto_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1554 entries, 0 to 1553
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   datebit       1554 non-null   datetime64[ns]
 1   openbit       1554 non-null   float64       
 2   highbit       1554 non-null   float64       
 3   lowbit        1554 non-null   float64       
 4   closebit      1554 non-null   float64       
 5   adj_closebit  1554 non-null   float64       
 6   volumebit     1554 non-null   int64         
 7   openeth       1554 non-null   float64       
 8   higheth       1554 non-null   float64       
 9   loweth        1554 non-null   float64       
 10  closeeth      1554 non-null   float64       
 11  adj_closeeth  1554 non-null   float64       
 12  volumeeth     1554 non-null   int64         
dtypes: datetime64[ns](1), float64(10), int64(2)
memory usage: 170.0 KB


After checking, there are no null values since the non-null count for each column = the total entries i.e 1554

In [267]:
#Dropping the unnecessary columns 'closebit' and 'closeeth' since we can use 'adj_closebit' and 'adj_closeeth' which more conclusive figures.

crypto_data.drop(["closebit" , "closeeth"], axis = 1, inplace = True)
crypto_data.head(10)

datebit       openbit       highbit  ...      loweth  adj_closeeth   volumeeth
0 2017-11-21   8205.740234   8348.660156  ...  350.692993    360.401001   949912000
1 2017-11-22   8077.950195   8302.259766  ...  360.147003    380.652008   800819008
2 2017-11-23   8232.379883   8267.400391  ...  376.088013    410.165985  1845680000
3 2017-11-24   8074.020020   8374.160156  ...  402.757996    474.911011  2292829952
4 2017-11-25   8241.709961   8790.919922  ...  461.053009    466.276001  1422080000
5 2017-11-26   8789.040039   9522.929688  ...  451.605988    471.329987  1197779968
6 2017-11-27   9352.719727   9818.349609  ...  468.484985    480.355011  1396480000
7 2017-11-28   9823.429688  10125.700195  ...  466.346985    472.902008  1346499968
8 2017-11-29  10077.400391  11517.400391  ...  425.071014    427.523010  2675940096
9 2017-11-30   9906.790039  10801.000000  ...  401.243011    447.114014  1903040000

[10 rows x 11 columns]

In [270]:
#Check Consistency and remove duplicated date records
crypto_data.drop_duplicates(subset=['datebit'])

datebit       openbit  ...  adj_closeeth    volumeeth
0    2017-11-21   8205.740234  ...    360.401001    949912000
1    2017-11-22   8077.950195  ...    380.652008    800819008
2    2017-11-23   8232.379883  ...    410.165985   1845680000
3    2017-11-24   8074.020020  ...    474.911011   2292829952
4    2017-11-25   8241.709961  ...    466.276001   1422080000
...         ...           ...  ...           ...          ...
1549 2022-02-17  43937.070313  ...   2881.481934  15860206214
1550 2022-02-18  40552.132813  ...   2785.727539  15748173433
1551 2022-02-19  40026.023438  ...   2763.701172   9774183169
1552 2022-02-20  40118.101563  ...   2628.648438  11641437834
1553 2022-02-21  38339.218750  ...   2701.658936  15016932352

[1554 rows x 11 columns]

In [271]:
#Exporting our new clean and merged dataset to a single csv file.
crypto_data.to_csv("crypto_dataset.csv", index=False)